### ISSUE TITLE
Weird error and delay pulling non-existent image from private repo

### ISSUE THREAD

**Issue by corrieb** [vmware/vic#2159]
In VIC 0.5.5, if you try to pull a non-existent image from a private repo, it waits 10 seconds and then comes back with:
```
root@office1-sfo2-dhcp270:~/vic# time docker --tls pull 10.118.67.164/vmworld/foo
Using default tag: latest
Pulling from vmworld/foo
Error while pulling image manifest: download cancelled during retry delay

real	0m10.106s
```
With regular Docker, this is the result
```
root@office1-sfo2-dhcp270:~# time docker pull 10.118.67.164/vmworld/foo
Using default tag: latest
Pulling repository 10.118.67.164/vmworld/foo
Error: image vmworld/foo:latest not found

real	0m0.054s
```
VIC shouldn't delay and we should report a more friendly message like the Docker one

**Comment by anchal-agrawal** [comment]
Related/dup of #2117

**Comment by anchal-agrawal** [comment]
This should have been fixed in #2258. I'm getting this error message now:

```
$ docker -H 192.168.77.130:2376 --tls pull 192.168.77.163:5000/fakeimage
Using default tag: latest
Failed to obtain OAuth endpoint: fakeimage:latest does not exist at http://192.168.77.163:5000/v2/
```

Matching docker's error message is a simple change. I'll create another issue for it. @corrieb, please re-open this if you come across the original issue again.
